### PR TITLE
Remove 'var' declarations and reduce 'if' statements in updater.js

### PR DESCRIPTION
**Pull request by ygnoh**
- Remove `var` declarations. `var` is obsolete and `let` and `const` are more intuitive and safe.
- Reduce unnecessary `if` statements.

### CODE REVIEW

**Comment by greysteil** [comment]
Thanks @ygnoh.

@feelepxyz - this is all yours to review!

File: helpers/npm/lib/updater.js [hunk 1]
```diff
@@ -68,20 +68,17 @@ async function updateDependencyFiles(
...
+  const source = (requirements.find(req => req.source) || {}).source;
 
   if (source && source.type === "git") {
-    var originalVersion = ((oldLockfile["dependencies"] || {})[depName] || {})[
+    let originalVersion = ((oldLockfile["dependencies"] || {})[depName] || {})[
       "version"
     ];
-    originalVersion = originalVersion || `${source.url}#ref`;
+
     if (!originalVersion.includes("#")) {
```
Comment (feelepxyz) [inline helpers/npm/lib/updater.js]: @ygnoh looks like `originalVersion` could be undefined here so `.includes()` will throw, could you guard against this here? `if (!originalVersion || !originalVersion.includes("#")) {...`

Comment (feelepxyz) [inline helpers/npm/lib/updater.js]:
```suggestion
     if (!originalVersion || !originalVersion.includes("#"))
```

Comment (ygnoh) [reply helpers/npm/lib/updater.js]: Thank you! It would be `undefined`. I think `oldLockFile` always has some value, not `undefined`. I will change it!

**Comment by ygnoh** [comment]
@feelepxyz I just squashed all commits into one! Please check

**Comment by feelepxyz** [comment]
@ygnoh nice one! Yeah looks like the test failures are unrelated to your changes, will have a look when I get moment.

**Comment by greysteil** [comment]
That failure is mine - sorry about that! Doesn't need to block this 🙂

**Pull request merged**